seisidd/experiment.py

```diff
@@ -321,10 +425,46 @@ def tomo_scan(self, cfg):
         )
         n_projections = len(angs)
         cfg['tomo']['n_projections'] = n_projections
-        total_images  = n_white + n_projections + n_white + n_dark
+        cfg['tomo']['total_images']  = n_white + n_projections + n_white + n_dark
         fp = cfg['output']['filepath']
         fn = cfg['output']['fileprefix']
-        
+
+        # consider adding an extra step to:
+        #   Perform energy calibration, set intended attenuation
+        #   set the lenses, change the intended slit size
+        #   prime the control of FS
+
+        #############################################
+        ## step 0.1: check and set beam parameters ##
+        #############################################
+        # set slit sizes
+        # These are the 1-ID-E controls
+        #   epics_put("1ide1:Kohzu_E_upHsize.VAL", ($1), 10) ##
+        #   epics_put("1ide1:Kohzu_E_dnHsize.VAL", (($1)+0.1), 10) ##
+        #   epics_put("1ide1:Kohzu_E_upVsize.VAL", ($2), 10) ## VERT SIZE
+        #   epics_put("1ide1:Kohzu_E_dnVsize.VAL", ($2)+0.1, 10) ##
+        _beam_h_size    =   cfg['tomo']['beamsize_h']
+        _beam_v_size    =   cfg['tomo']['beamsize_v']
+        yield from bps.mv(beam.s1.h_size, _beam_h_size          )
+        yield from bps.mv(beam.s1.v_size, _beam_v_size          )
+        yield from bps.mv(beam.s2.h_size, _beam_h_size + 0.1    )       # add 0.1 following 1ID convention
+        yield from bps.mv(beam.s2.v_size, _beam_v_size + 0.1    )       # to safe guard the beam?
+
+        # set attenuation
+        _attenuation = cfg['tomo']['attenuation']
+        yield from bps.mv(beam.att.att_level, _attenuation)
+
+        # check energy
+        # need to be clear what we want to do here
+        _energy_foil = cfg['tomo']['energyfoil']
+        yield from bps.mv(beam.foil, _energy_foil)      # need to complete this part in beamline.py
```

Comment:
It seems to me that the beamsize and beam energy should be pre-configured in the notebook.  Asking RE to handle this part feels dangerous to me.

Reply:
Agreed